lms/djangoapps/support/views/program_enrollments.py

```diff
@@ -297,3 +244,131 @@ def _get_org_key_names(self):
         ).select_related('organization')
 
         return [saml_provider.organization.short_name for saml_provider in saml_providers]
+
+
+class ProgramEnrollmentsInspectorView(ProgramEnrollmentInspector, View):
+    """
+    The view to search and display the program enrollments
+    information of a learner.
+    """
+    exclude_from_schema = True
+    CONSOLE_TEMPLATE_PATH = 'support/program_enrollments_inspector.html'
+
+    @method_decorator(require_support_permission)
+    def get(self, request):
+        """
+        Based on the query string parameters passed through the GET request
+        Search the data store for information about ProgramEnrollment and
+        SSO linkage with the user.
+        """
+        search_error = ''
+        edx_username_or_email = request.GET.get('edx_user', '').strip()
+        org_key = request.GET.get('org_key', '').strip()
+        external_user_key = request.GET.get('external_user_key', '').strip()
+        learner_program_enrollments = {}
+        saml_providers_with_org_key = self._get_org_keys_and_idps()
+        selected_provider = None
+        if org_key:
+            selected_provider = saml_providers_with_org_key.get(org_key)
+        if edx_username_or_email:
+            learner_program_enrollments, search_error = self._get_account_info(
+                edx_username_or_email,
+                selected_provider,
+            )
+        elif org_key and external_user_key:
+            learner_program_enrollments = self._get_external_user_info(
+                external_user_key,
+                org_key,
+                selected_provider,
+            )
+            if not learner_program_enrollments:
+                search_error = 'No user found for external key {} for institution {}'.format(
+                    external_user_key, org_key
+                )
+        elif not org_key and not external_user_key:
+            # This is initial rendering state.
+            pass
```

Comment:
we may move this to start

Reply:
This is refactored from the old view. This will be deprecated eventually so it doesn't matter.